File: spec/factories/api_key.rb
# typed: true

FactoryBot.define do
  factory :api_key do
    name { Faker::ProgrammingLanguage.name }
    token { SecureRandom.hex }
    token_digest { SecureRandom.hex }
    last_used_on_utc { nil }
    bearer_type { "User" }

    association :bearer, factory: :user

    initialize_with {
      new({
        name:,
        token:,
        token_digest:,
        last_used_on_utc:,
        bearer_type:
      })
    }
  end
end

File: go-backend/models/api_key.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// APIKey model
type APIKey struct {
	gorm.Model
	BearerType    string `gorm:"not null"`
	LastUsedOnUTC time.Time
	Name          string
	TokenDigest   string `gorm:"not null;unique"`
	BearerID      uint   `gorm:"not null"`
}
